static/specification.html

```diff
@@ -138,605 +336,555 @@ <h2>
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h4>
+            `monetization-src` Post-request check
+          </h4>
+          <p>
+            This directive's post-request check is as follows:
+          </p>
+          <p>
+            Given a [=request=] (|request|) and a [=violation/policy=]
+            (|policy|):
+          </p>
+          <ol>
+            <li>Let |name| be the result of executing "Get the effective
+            directive for request" on |request|.
+            </li>
+            <li>If the result of executing "Should fetch directive execute" on
+            |name|, `monetization-src` and |policy| is "No", return "Allowed".
+            </li>
+            <li>If the result of executing "Does response to request match
+            source list?" on |response|, |request|, this directive's value, and
+            |policy|, is "Does Not Match", return "Blocked".
+            </li>
+            <li>Return "Allowed".
+            </li>
+          </ol>
+        </section>
       </section>
     </section>
+    <section>
+      <h2>
+        Security considerations
+      </h2>
+      <p>
+        It is RECOMMENDED that a user agent provide some UI or indicator that
+        allows the user to know when [=monetization=] is possible and/or when
+        [=monetization=] is occurring. Providing such a UI allows the users to
+        retain control of the monetization process by taking action (e.g., stop
+        or start monetization of a particular site if they wish to do so).
+      </p>
+      <p>
+        As [=payment end-points=] are generally provided as a service (e.g.,
+        Uphold), a <abbr title="Cross Site Scripting">XSS</abbr> attack could
+        inject malicious [=payment end-points=] into a page that uses the same
+        service. To mitigate such an attack, it is RECOMMENDED that developers:
+      </p>
+      <ul>
+        <li>host a [=payment end-point=] on their own servers and proxy [[Open
+        Payments]] requests on the server as needed.
+        </li>
+        <li>Set the `monetization-src` CSP directive to restrict requests to
+        origins they control and trust to host [=payment end-points=].
+        </li>
+      </ul>
+    </section>
+    <section>
+      <h2>
+        Privacy considerations
+      </h2>
+      <p>
+        Web Monetization is designed to be privacy-preserving: The user agent
+        does not send any data to the [=monetization provider=]. Instead, it
+        requests data from the [=monetization provider=] without ever revealing
+        the URL of the web page the user is currently browsing.
+      </p>
+      <p>
+        Further, the user agent gets the payment information from the [=payment
+        end-point=] to establish the [=payment session=]. This also ensures the
+        [=monetization provider=] doesn't have access to a user's browsing
+        history or to the [=payment end-point=].
```

Comment:
It seems both outgoing and incoming payments are POST created resources 
https://docs.openpayments.guide/reference/create-outgoing-payment

Can a link's GET request suffice to get Open Payment connection details? 
Or some other step that requires the payment pointer (and perhaps some other credentials) is necessary ?

Reply:
The privacy here is achieved by separating the processing of the payment pointer and the sending of the payment.

The user agent is the only thing that should process the payment pointer.

The WM Provider needs to send the payment and provide data back to the browser about when it is sending a payment, how much it sent and also how the website can verify the payment.

The interface between the browser and the WM Provider is not defined in this spec. How a browser chooses to allow WM Providers to integrate is up to them. Ideally it would be through the extensions system but it could be something else like the Payment Handler API.

OP was designed the way it is (create an incoming payment first) specifically to support this privacy-enhancing separation. i.e. The browser creates the incoming payment and only provides the connection URL to the WM Provider.

A browser does need to authenticate to create an incoming payment but it can dynamically register (i.e. no user interaction required) as a client with the wallet that issued the payment pointer which is a once off to prevent DDoS attacks.